docs/design/query-caching.md

```diff
@@ -0,0 +1,224 @@
+
+## Configuration Changes
+
+Entity object have the following top-level properties: source, graphql, rest, permissions, relationships, mappings. Caching will introduce a new top-level property for entities to enable entity-level granularity in cache settings.
+
+```json
+{
+  "entities": {
+    "EntityA": {
+      "cache": {
+        "enabled": true,
+        "ttl": 123
+      }
+    }
+  }
+}
+```
+
+| Property | JSON Data Type | Default Value | Description Value | Required |
+|----------|------------|-------------|---------------|----------|
+| enabled  | Boolean | false | Whether caching is enabled for read requests on this entity. |  Yes |
+| ttl      | Number | 30 | Number of seconds a cache entry is valid before cache eviction. Min: 1s Max: 24hrs/1440s |  Yes |
+
+Entity configuration does not require you to set the **cache** property because caching is disabled by default. When you enable caching explicitly by setting **enabled** to true, you must also define the **ttl** property with an integer value. When evaluating the behavior of caching an entity, you can set the **enabled** property to false instead of removing the entire caching section from the entity's configuration section.
+
+**Note:** You should consider enabling caching on an entity when existing records are rarely modified and the entity is frequently accessed.
+
+While the **ttl** property has a default value of 30 seconds, you should specify a value most applicable for your use case. 
+- A higher **ttl** value may be appropriate for entities which are rarely modified and frequently accessed.
+- A lower **ttl** value may be appropriate for entities which may change on occassion and are frequentiy accessed. 
+
+## Endpoint Behavior
+
+### GraphQL
+
+**Nested queries** result in joins between multiple database objects which transitively implies joins between multiple entities. Consider the following GraphQL query whose top-level query references the `Book` object type and the nested query references the `Publisher` object type:
+
+```graphql
+query NestedEntityQuery{
+  book_by_pk (id: 1) {
+    id,
+    title,
+    publishers {
+      id,
+      name
+    }
+  }
+}
+```
+
+Because DAB resolves the request into a single database query, DAB is unable to query the cache for one half the request and query the database for the other half. 
+**DAB will only honor the cache configuration for the top-level referenced entity.** The developer will be responsible for ensuring their cache settings are consistently set across related entities to experience caching behavior consistent across entities.
```

Comment:
What would be the guidance if the top level entity is mostly READ only but the related nested entities can be modified frequently?

Reply:
Then we'd need to suggest that when using GraphQL, you need to determine whether caching is right in that circumstance. 

Or we can look at skipping the cache altogether when nested entities. are present.  Currently, I'm proposing caching the entire SQL query because otherwise we'd need to modify our GraphQL resolver code to know when or when not to generate sql text and instead fetch from the cache, and then proceed to know how to piece cache results with non-cached results. DAB hasn't really been built to accommodate that which may require a much larger refactor / different design.

Reply:
makes sense, we should incorporate the guidance on which entities to cache when we provide documentation updates for the caching feature.